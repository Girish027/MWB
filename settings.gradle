rootProject.name = 'nl-tools-ui-2.6.2'