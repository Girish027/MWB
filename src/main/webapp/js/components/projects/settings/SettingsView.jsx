import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  LegacyRow as Row,
  LegacyColumn as Column,
  TextField,
  Tabs,
  VerticalNav,
  VerticalNavItem,
} from '@tfs/ui-components';
import Constants from 'constants/Constants';

export class SettingsView extends Component {
  constructor(props) {
    super(props);

    this.getModelLevelMonitoringSection = this.getModelLevelMonitoringSection.bind(this);
    this.getNodeLevelMonitoringSection = this.getNodeLevelMonitoringSection.bind(this);
    this.getNodeLevelMonitoringTabs = this.getNodeLevelMonitoringTabs.bind(this);
    this.renderUniversalTabItems = this.renderUniversalTabItems.bind(this);
    this.renderEachNodeTabItems = this.renderEachNodeTabItems.bind(this);
    this.onNavItemSelected = this.onNavItemSelected.bind(this);
    this.state = {
      activeNodeLevelTab: Constants.UNIVERSAL_TAB,
      selectedNavItem: 'Node 1',
      percentageForADay: '30%',
      percentageOnConsecutiveDays: '0%',
    };
  }

  onTabSelected = (selectedTab, selectedTabIndex) => {
    let tab = '';
    switch (selectedTabIndex) {
    case 0:
      tab = Constants.UNIVERSAL_TAB;
      break;
    case 1:
      tab = Constants.EACH_NODE_TAB;
      break;
    default:
      tab = Constants.UNIVERSAL_TAB;
      break;
    }
    this.setState({
      activeNodeLevelTab: tab,
    });
  }

  renderUniversalTabItems() {
    return (
      <Row>
        <div className="EscalationRate" style={{ paddingTop: '15px' }}>
          {Constants.ESCALATION_RATE_TITLE}
        </div>
        <div className="EscalationRateDesc">
          {Constants.ESCALATION_RATE_DESCRIPTION}
        </div>
        <div className="Threshold">
          {Constants.SET_THRESHOLD_TEXT_FOR_A_DAY}
          <TextField
            name="thresholdValue"
            defaultValue={this.state.percentageForADay}
            onChange={() => {}}
            styleOverride={{
              width: '45px',
              input: {
                width: '100%',
                height: '23px',
              },
            }}
          />
          {Constants.PERCENTAGE_FOR_A_DAY}
        </div>
      </Row>
    );
  }

  onNavItemSelected(itemName) {
    if (itemName) {
      return (
        <div>
          <div className="EscalationRate">
            {Constants.ESCALATION_RATE_TITLE}
          </div>
          <div className="Threshold">
            {Constants.SET_THRESHOLD_TEXT_FOR_A_DAY}
            <TextField
              name="thresholdValue"
              defaultValue={this.state.percentageForADay}
              onChange={() => {}}
              styleOverride={{
                width: '45px',
                input: {
                  width: '100%',
                  height: '23px',
                },
                helpText: {
                  height: '0px',
                },
              }}
            />
            {Constants.PERCENTAGE_FOR_A_DAY}
          </div>
          <div className="DefaultThreshold">
            {Constants.DEFAULT_THRESHOLD}
          </div>
        </div>
      );
    }
  }

  onClickNavItem = (navItem) => {
    this.setState({
      selectedNavItem: navItem[0],
    });
  }

  renderEachNodeTabItems() {
    const { selectedNavItem } = this.state;
    return (
      <Row>
        <Column
          size={2}
          styleOverride={{
            height: 'auto',
            borderRight: '1px solid #CCC',
            position: 'relative',
          }}
        >
          <VerticalNav
            selectedItems={[this.state.selectedNavItem]}
            onSelection={this.onClickNavItem}
            styleOverride={{
              boxShadow: 'none',
              overflowY: 'auto',
              padding: '20px 0px',
              marginLeft: -6,
            }}
          >
            <VerticalNavItem
              name="Node 1"
            >
              Node 1
            </VerticalNavItem>
            <VerticalNavItem
              name="Node 2"
            >
              Node 2
            </VerticalNavItem>
          </VerticalNav>
        </Column>
        <Row>
          <Column
            size={9}
            styleOverride={{
              padding: '25px 0 0 25px',
              overflowY: 'auto',
            }}
          >
            {this.onNavItemSelected(selectedNavItem)}
          </Column>
        </Row>
      </Row>
    );
  }

  getNodeLevelMonitoringTabs() {
    const { activeNodeLevelTab } = this.state;
    const tabs = [Constants.UNIVERSAL_TAB, Constants.EACH_NODE_TAB];
    const tabPanels = [this.renderUniversalTabItems(), this.renderEachNodeTabItems()];

    let selectedTabIndex = 0;
    switch (activeNodeLevelTab) {
    case Constants.UNIVERSAL_TAB:
      selectedTabIndex = 0;
      break;
    case Constants.EACH_NODE_TAB:
      selectedTabIndex = 1;
      break;
    default:
      selectedTabIndex = 0;
      break;
    }

    return (
      <div className="tabs">
        <Tabs
          tabs={tabs}
          tabPanels={tabPanels}
          onTabSelected={this.onTabSelected}
          align="left"
          selectedIndex={selectedTabIndex}
          styleOverride={{
            tabItem: {
              padding: '23px 20px 21px',
            },
            tabContainer: {
              borderTop: 'none',
            },
          }}
        />
      </div>
    );
  }

  getModelLevelMonitoringSection() {
    return (
      <Row>
        <div className="ModelLevelMonitoringTitle">
          {Constants.MODEL_LEVEL_MONITORING_THRESHOLD_TITLE}
        </div>
        <hr color="#bcc1c7" />
        <div className="EscalationRate">
          {Constants.ESCALATION_RATE_TITLE}
        </div>
        <div className="Threshold">
          {Constants.SET_THRESHOLD_TEXT_FOR_A_DAY}
          <TextField
            name="thresholdValue"
            defaultValue={this.state.percentageForADay}
            onChange={() => {}}
            styleOverride={{
              width: '45px',
              input: {
                width: '100%',
                height: '23px',
              },
              helpText: {
                height: '0px',
              },
            }}
          />
          {Constants.PERCENTAGE_FOR_A_DAY}
        </div>
        <div className="DefaultThreshold">
          {Constants.DEFAULT_THRESHOLD}
        </div>
        <div className="Threshold">
          {Constants.SET_THRESHOLD_TEXT_FOR_CONSECUTIVE_DAYS}
          <TextField
            name="thresholdValue"
            defaultValue={this.state.percentageOnConsecutiveDays}
            onChange={() => {}}
            styleOverride={{
              width: '45px',
              input: {
                width: '100%',
                height: '23px',
              },
              helpText: {
                height: '0px',
              },
            }}
          />
          {Constants.PERCENTAGE_ON_CONSECUTIVE_DAYS}
        </div>
      </Row>
    );
  }

  getNodeLevelMonitoringSection() {
    return (
      <div>
        <div className="NodeLevelMonitoringTitle">
          {Constants.NODE_LEVEL_MONITORING_THRESHOLD_TITLE}
        </div>
        <hr color="#bcc1c7" style={{ marginBottom: '0px' }} />
        <div>
          {this.getNodeLevelMonitoringTabs()}
        </div>
      </div>
    );
  }

  render() {
    return (
      <div id="SettingsTab">
        <Row>
          {this.getModelLevelMonitoringSection()}
          {this.getNodeLevelMonitoringSection()}
        </Row>
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({});
const mapDispatchToProps = dispatch => ({
  dispatch,
});

export default connect(mapStateToProps, mapDispatchToProps)(SettingsView);
