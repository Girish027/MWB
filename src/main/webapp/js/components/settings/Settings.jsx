import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  LegacyRow as Row,
  LegacyColumn as Column,
  VerticalNav,
  VerticalNavItem,
  Button,
  Dialog,
  Checkbox,
  TextField,
} from '@tfs/ui-components';
import * as preferencesActions from 'state/actions/actions_preferences';
import IconButton from 'components/IconButton';
import Constants from 'constants/Constants';
import TensorFlowIcon from '../Icons/TensorFlowIcon';
import NGramIcon from '../Icons/NGramIcon';

export class Settings extends Component {
  constructor(props) {
    super(props);

    this.getStoredTechnology = this.getStoredTechnology.bind(this);
    this.getConfirmationDialog = this.getConfirmationDialog.bind(this);
    this.onClickConfirm = this.onClickConfirm.bind(this);
    this.onClickCancel = this.onClickCancel.bind(this);
    this.onClickCheckbox = this.onClickCheckbox.bind(this);
    this.onChange = this.onChange.bind(this);
    this.getTensorFlowTile = this.getTensorFlowTile.bind(this);
    this.onClickTensorFlowTile = this.onClickTensorFlowTile.bind(this);
    this.getNGramTile = this.getNGramTile.bind(this);
    this.onClickNGramTile = this.onClickNGramTile.bind(this);
    this.onClickNavItem = this.onClickNavItem.bind(this);

    this.props = props;
    this.state = {
      selectedNavItem: Constants.NAV_MODEL_TECHNOLOGIES,
      previousTechnology: Constants.MODEL_TECHNOLOGY.N_GRAM,
      currentTechnology: '',
      showDialog: false,
      isChecked: false,
      componentMounted: false,
      localUpdate: false,
      percentageInADay: 32,
    };
  }

  componentDidMount() {
    this.setState({
      componentMounted: true,
    });
  }

  componentDidUpdate(prevProps, prevState) {
    const prevClientId = prevProps.preferences.clientId;
    const clientId = this.props.preferences.clientId;
    const prevCompMounted = prevState.componentMounted;
    const compMounted = this.state.componentMounted;
    if (prevClientId !== clientId || prevCompMounted !== compMounted) {
      const { clientId } = this.props.preferences;
      this.props.getVectorizer();
      this.props.getTechnology({
        clientId: clientId || this.props.header.client.id,
      });
    }
  }

  onClickNavItem(navItem) {
    this.setState({
      selectedNavItem: navItem[0],
    });
  }

  getStoredTechnology() {
    const { currentTechnology, localUpdate } = this.state;
    const { technology } = this.props.preferences;
    if (technology && technology !== '' && technology !== currentTechnology && !localUpdate) {
      this.setState({
        previousTechnology: currentTechnology,
        currentTechnology: technology,
        localUpdate: false,
      });
    }
  }

  onClickConfirm() {
    const clientId = this.props.header.client.id;
    const { currentTechnology, previousTechnology, isChecked } = this.state;
    this.props.addOrUpdateTechnology({
      clientId,
      technologyValue: currentTechnology,
      updateExistingModels: isChecked,
    });
    this.setState({
      showDialog: false,
      isChecked: false,
      previousTechnology: previousTechnology !== currentTechnology ? currentTechnology : previousTechnology,
    });
  }

  onClickCancel() {
    const { previousTechnology } = this.state;
    this.setState({
      showDialog: false,
      isChecked: false,
      currentTechnology: previousTechnology,
      localUpdate: false,
    });
  }

  onClickCheckbox(event) {
    this.setState({
      isChecked: event.checked,
    });
  }

  onChange(event) {
    const { value } = event.target;
    this.setState({
      percentageInADay: value,
    });
  }

  getConfirmationDialog() {
    const { showDialog, isChecked } = this.state;
    let result;
    if (showDialog) {
      result = (
        <Dialog
          isOpen
          headerChildren={Constants.SAVE_DIALOG_TITLE}
          size="small"
          closeIconVisible={false}
          okChildren="CONFIRM"
          cancelChildren="CANCEL"
          onClickOk={this.onClickConfirm}
          onClickCancel={this.onClickCancel}
        >
          <div
            style={{
              textAlign: 'left',
              lineHeight: '25px',
              fontSize: '14px',
              paddingBottom: '25px',
            }}
          >
            {Constants.SAVE_DIALOG_CONTENT}
          </div>
          <Checkbox
            checked={isChecked}
            value
            label={Constants.SAVE_DIALOG_CHECKBOX}
            onChange={(event) => this.onClickCheckbox(event.target)}
          />
        </Dialog>
      );
    }
    return result;
  }

  onClickTensorFlowTile() {
    const { currentTechnology } = this.state;
    this.setState({
      previousTechnology: currentTechnology,
      currentTechnology: Constants.MODEL_TECHNOLOGY.USE,
      localUpdate: true,
    });
  }

  onClickNGramTile() {
    const { currentTechnology } = this.state;
    this.setState({
      previousTechnology: currentTechnology,
      currentTechnology: Constants.MODEL_TECHNOLOGY.N_GRAM,
      localUpdate: true,
    });
  }

  getTensorFlowIcon = () => {
    const { latestTechnology } = this.props.preferences;
    const version = latestTechnology && latestTechnology.version || '';
    return (
      <TensorFlowIcon
        version={version}
        width={180}
        height={200}
      />
    );
  }

  getTensorFlowTile() {
    const { currentTechnology } = this.state;
    const styles = currentTechnology === Constants.MODEL_TECHNOLOGY.USE ? {
      border: '2px solid #000000',
    } : {
      border: '2px solid #dedede',
    };
    return (
      <IconButton
        onClick={this.onClickTensorFlowTile}
        icon={this.getTensorFlowIcon}
        height={200}
        width={180}
        styleOverride={{
          marginRight: '25px',
          borderRadius: '5px',
          ':hover': {
            border: '2px solid #004c97',
          },
          ...styles,
        }}
      />
    );
  }

  getNGramTile() {
    const { currentTechnology } = this.state;
    const styles = currentTechnology === Constants.MODEL_TECHNOLOGY.N_GRAM ? {
      border: '2px solid #000000',
    } : {
      border: '2px solid #dedede',
    };
    return (
      <IconButton
        onClick={this.onClickNGramTile}
        icon={NGramIcon}
        height={200}
        width={180}
        styleOverride={{
          borderRadius: '5px',
          ':hover': {
            border: '2px solid #004c97',
          },
          ...styles,
        }}
      />
    );
  }

  onNavItemSelected(itemName) {
    let result;
    switch (itemName) {
    case Constants.NAV_MODEL_TECHNOLOGIES:
      this.getStoredTechnology();
      result = (
        <div>
          <div
            style={{
              color: '#333333',
              fontSize: '15px',
              lineHeight: '30px',
              fontWeight: 'bold',
              marginBottom: '4px',
            }}
          >
            {Constants.MODEL_TECHNOLOGY_TITLE}
          </div>
          <div
            style={{
              color: '#9B9B9B',
              fontSize: '14px',
              lineHeight: '13px',
              fontWeight: 'normal',
            }}
          >
            {Constants.MODEL_TECHNOLOGY_DESCRIPTION}
          </div>
          <hr />
          {this.getTensorFlowTile()}
          {this.getNGramTile()}
          <div
            style={{
              paddingLeft: '800px',
              paddingTop: '40px',
            }}
          >
            <Button
              type="primary"
              name="save-as-default-button"
              onClick={() => {
                this.setState({
                  showDialog: true,
                });
              }}
              styleOverride={{
                paddingLeft: '25px',
                paddingRight: '25px',
              }}
            >
              SAVE AS DEFAULT
            </Button>
          </div>
          <hr />
        </div>
      );
      break;
    case Constants.NAV_MODEL_MONITORING:
      result = (
        <div>
          <div
            style={{
              color: '#333333',
              fontSize: '15px',
              lineHeight: '30px',
              fontWeight: 'bold',
              marginBottom: '4px',
            }}
          >
            {Constants.MODEL_MONITORING_CONFIG_TITLE}
          </div>
          <div
            style={{
              color: '#9B9B9B',
              fontSize: '14px',
              lineHeight: '13px',
              fontWeight: 'normal',
            }}
          >
            {Constants.MODEL_MONITORING_DESCRIPTION}
          </div>
          <hr />
          <div
            style={{
              color: '#333333',
              fontSize: '15px',
              lineHeight: '30px',
              fontWeight: 'bold',
              marginBottom: '4px',
            }}
          >
            {Constants.ESCALATION_RATE_TITLE}
          </div>
          <div
            className="Threshold"
            style={{
              color: '#4A4A4A',
              fontSize: '15px',
              lineHeight: '13px',
              fontWeight: 'normal',
            }}
          >
            {Constants.SET_THRESHOLD_TEXT_FOR_A_DAY}
            <input
              name="thresholdValue"
              type="number"
              min="1"
              max="100"
              value={this.state.percentageInADay}
              onChange={this.onChange}
              ref="input"
            />
            {Constants.PERCENTAGE_IN_A_DAY}
          </div>
          <br />
          <hr />
          <div
            style={{
              paddingLeft: '800px',
              paddingTop: '20px',
            }}
          >
            <Button
              type="primary"
              name="save-item"
              styleOverride={{
                paddingLeft: '25px',
                paddingRight: '25px',
              }}
            >
                  SAVE
            </Button>
          </div>
        </div>
      );
      break;
    default:
      // Do nothing
    }
    return result;
  }

  render() {
    const { selectedNavItem } = this.state;
    return (
      <div id="Settings">
        <Row>
          <Column
            size={2}
            styleOverride={{
              height: 'calc(100vh - 101px)',
              borderRight: '1px solid #CCC',
              position: 'relative',
            }}
          >
            <VerticalNav
              selectedItems={[this.state.selectedNavItem]}
              onSelection={this.onClickNavItem}
              styleOverride={{
                boxShadow: 'none',
                overflowY: 'auto',
                padding: '20px 0px',
                marginLeft: -6,
                marginTop: '-20px',
              }}
            >
              <VerticalNavItem
                styleOverride={{ lineHeight: '60px', fontSize: '15px' }}
                name={Constants.NAV_MODEL_TECHNOLOGIES}
              >
                Model Technologies
              </VerticalNavItem>
              <VerticalNavItem
                styleOverride={{ lineHeight: '60px', fontSize: '15px' }}
                name={Constants.NAV_MODEL_MONITORING}
              >
                Model Monitoring
              </VerticalNavItem>
            </VerticalNav>
          </Column>
          <Row>
            <Column
              size={9}
              styleOverride={{
                padding: '25px 0 0 25px',
                overflowY: 'auto',
              }}
            >
              {this.onNavItemSelected(selectedNavItem)}
            </Column>
            {this.getConfirmationDialog()}
          </Row>
        </Row>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  preferences: state.preferences,
  header: state.header,
});
const mapDispatchToProps = dispatch => ({
  getTechnology: ({ clientId }) => {
    dispatch(preferencesActions.getTechnology({ clientId }));
  },
  addOrUpdateTechnology: ({ clientId, technologyValue, updateExistingModels }) => {
    dispatch(preferencesActions.addOrUpdateTechnology({ clientId, technologyValue, updateExistingModels }));
  },
  getVectorizer: () => {
    dispatch(preferencesActions.getVectorizer());
  },
  dispatch,
});

export default connect(mapStateToProps, mapDispatchToProps)(Settings);
