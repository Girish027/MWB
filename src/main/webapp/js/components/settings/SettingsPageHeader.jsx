import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import breadcrumbSegments from 'components/AppHeader/breadcrumbSegments';
import Constants from 'constants/Constants';
import {
  PageHeader,
  ToastNotification,
  Xmark,
  Button,
} from '@tfs/ui-components';
import { stopShowingServerMessage, changeRoute } from 'state/actions/actions_app.js';

export default class SettingsPageHeader extends Component {
  constructor(props) {
    super(props);

    this.props = props;
    this.types = Constants.NOTIFICATION.types;
  }

  closeNotification = () => {
    const { dispatch } = this.props;
    dispatch(stopShowingServerMessage());
  }

  getNotification = () => {
    const { app: { notificationType, serverMessage } = {} } = this.props;

    let notification = null;
    if (notificationType && serverMessage) {
      notification = (
        <ToastNotification
          type={notificationType}
          styleOverride={{
            top: '4px',
            width: '40%',
            display: 'inline-block',
            padding: '0px 8px',
            fontSize: '13px',
            fontWeight: 'normal',
            lineHeight: '20px',
          }}
        >
          {serverMessage}
          <Button
            type="flat"
            onClick={this.closeNotification}
            styleOverride={{
              marginLeft: '10px',
              ':hover': {
                boxShadow: 'none',
              },
              ':focus': {
                outline: 'none',
              },
            }}
          >
            <Xmark fill={notificationType === this.types.error ? '#c81919' : '#00ac86'} />
          </Button>
        </ToastNotification>
      );
    }
    return notification;
  }

  render() {
    const {
      selectedProject, selectedClient,
      match, history, itsURL,
    } = this.props;

    const breadcrumbs = breadcrumbSegments(match, {
      selectedClient,
      selectedProject,
    }, history, itsURL);

    return (
      <PageHeader
        title={Constants.PREFERENCES}
        breadcrumb={breadcrumbs}
      >
        {this.getNotification()}
      </PageHeader>
    );
  }
}

SettingsPageHeader.defaultProps = {
  history: {},
  selectedClient: {},
  selectedProject: {},
  match: {},
  itsURL: '',
  clientId: '',
  dispatch: () => {},
};

SettingsPageHeader.propTypes = {
  history: PropTypes.object,
  selectedClient: PropTypes.object,
  selectedProject: PropTypes.object,
  match: PropTypes.object,
  itsURL: PropTypes.string,
  dispatch: PropTypes.func,
};
