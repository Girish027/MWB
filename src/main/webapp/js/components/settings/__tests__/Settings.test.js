import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Settings } from 'components/settings/Settings';

describe('<Settings />', () => {
  const initialState = {
    selectedNavItem: 'Model Technologies',
    showDialog: true,
    isChecked: false,
    componentMounted: false,
    localUpdate: false,
    currentTechnology: 'n-gram',
    previousTechnology: 'use_large',
  };

  const testProps = {
    preferences: {
      technology: 'use_large',
    },
    getTechnology: jest.fn(() => 'called getTechnology'),
    addOrUpdateTechnology: jest.fn(() => 'called addOrUpdateTechnology'),
    getVectorizer: jest.fn(() => 'called getVectorizer'),
    header: {
      client: {
        id: '123',
      },
    },
    dispatch: jest.fn(),
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Settings
      {...testProps}
    />);
    wrapper.setState({
      ...initialState,
    });
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (props) => shallow(<Settings
    {...props}
  />);

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = getShallowWrapper(testProps);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('should match with basic props', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    beforeEach(() => {
      testProps.dispatch = jest.fn();
    });

    test('should call onClickNavItem - Model Technologies', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().onClickNavItem(['Model Technologies']);
      expect(wrapper.state().selectedNavItem).toBe('Model Technologies');
    });

    test('should call onClickNavItem - Model Monitoring', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().onClickNavItem(['Model Monitoring']);
      expect(wrapper.state().selectedNavItem).toBe('Model Monitoring');
    });

    test('should call getStoredTechnology', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().getStoredTechnology();
      expect(wrapper.state().currentTechnology).toBe('use_large');
      expect(wrapper.state().previousTechnology).toBe('');
    });

    test('should call onClickTensorFlowTile', () => {
      wrapper.setState({ ...initialState, currentTechnology: 'use_large', previousTechnology: 'n-gram' });
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().onClickTensorFlowTile();
      expect(wrapper.state().previousTechnology).toBe('use_large');
    });

    test('should call onClickNGramTile', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.setState({ ...initialState, currentTechnology: 'n-gram', previousTechnology: 'use_large' });
      wrapper.instance().onClickTensorFlowTile();
      expect(wrapper.state().previousTechnology).toBe('n-gram');
    });

    test('should call onClickCancel', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.setState({ ...initialState, currentTechnology: 'n-gram', previousTechnology: 'use_large' });
      wrapper.instance().onClickCancel();
      expect(wrapper.state().currentTechnology).toBe('use_large');
    });

    test('should call onClickCheckbox', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().onClickCheckbox({ checked: true });
      expect(wrapper.state().isChecked).toBe(true);
    });
  });
});
