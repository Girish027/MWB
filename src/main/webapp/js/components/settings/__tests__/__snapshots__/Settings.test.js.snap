// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> Snapshots should match with basic props 1`] = `
<div
  id="Settings"
>
  <Deprecated>
    <Deprecated
      size={2}
      styleOverride={
        Object {
          "borderRight": "1px solid #CCC",
          "height": "calc(100vh - 101px)",
          "position": "relative",
        }
      }
    >
      <withTheme(VerticalNav)
        onSelection={[Function]}
        selectedItems={
          Array [
            "Model Technologies",
          ]
        }
        styleOverride={
          Object {
            "boxShadow": "none",
            "marginLeft": -6,
            "marginTop": "-20px",
            "overflowY": "auto",
            "padding": "20px 0px",
          }
        }
      >
        <withTheme(VerticalNavItem)
          name="Model Technologies"
          styleOverride={
            Object {
              "fontSize": "15px",
              "lineHeight": "60px",
            }
          }
        >
          Model Technologies
        </withTheme(VerticalNavItem)>
        <withTheme(VerticalNavItem)
          name="Model Monitoring"
          styleOverride={
            Object {
              "fontSize": "15px",
              "lineHeight": "60px",
            }
          }
        >
          Model Monitoring
        </withTheme(VerticalNavItem)>
      </withTheme(VerticalNav)>
    </Deprecated>
    <Deprecated>
      <Deprecated
        size={9}
        styleOverride={
          Object {
            "overflowY": "auto",
            "padding": "25px 0 0 25px",
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "color": "#333333",
                "fontSize": "15px",
                "fontWeight": "bold",
                "lineHeight": "30px",
                "marginBottom": "4px",
              }
            }
          >
            Model Technology
          </div>
          <div
            style={
              Object {
                "color": "#9B9B9B",
                "fontSize": "14px",
                "fontWeight": "normal",
                "lineHeight": "13px",
              }
            }
          >
            Select one from the list below to set as a default technology.
          </div>
          <hr />
          <IconButton
            defaultColor="#004c97"
            disabled={false}
            disabledColor="#9B9B9B"
            focusedColor="#003467"
            height={200}
            icon={[Function]}
            onClick={[Function]}
            padding="1px"
            strokeColor="none"
            styleOverride={
              Object {
                ":hover": Object {
                  "border": "2px solid #004c97",
                },
                "border": "2px solid #dedede",
                "borderRadius": "5px",
                "marginRight": "25px",
              }
            }
            title=""
            width={180}
          />
          <IconButton
            defaultColor="#004c97"
            disabled={false}
            disabledColor="#9B9B9B"
            focusedColor="#003467"
            height={200}
            icon={[Function]}
            onClick={[Function]}
            padding="1px"
            strokeColor="none"
            styleOverride={
              Object {
                ":hover": Object {
                  "border": "2px solid #004c97",
                },
                "border": "2px solid #000000",
                "borderRadius": "5px",
              }
            }
            title=""
            width={180}
          />
          <div
            style={
              Object {
                "paddingLeft": "800px",
                "paddingTop": "40px",
              }
            }
          >
            <withTheme(Button)
              name="save-as-default-button"
              onClick={[Function]}
              styleOverride={
                Object {
                  "paddingLeft": "25px",
                  "paddingRight": "25px",
                }
              }
              type="primary"
            >
              SAVE AS DEFAULT
            </withTheme(Button)>
          </div>
          <hr />
        </div>
      </Deprecated>
      <withTheme(DialogComponent)
        cancelChildren="CANCEL"
        closeIconVisible={false}
        headerChildren="Are you sure"
        isOpen={true}
        okChildren="CONFIRM"
        onClickCancel={[Function]}
        onClickOk={[Function]}
        size="small"
      >
        <div
          style={
            Object {
              "fontSize": "14px",
              "lineHeight": "25px",
              "paddingBottom": "25px",
              "textAlign": "left",
            }
          }
        >
          Changing the default value will not affect the already existing models, instead the new models will use the default technology. You can always change the technology while creating or editing a model version
        </div>
        <withTheme(Checkbox)
          checked={false}
          label="Update model technology of all the existing models to this default"
          onChange={[Function]}
          value={true}
        />
      </withTheme(DialogComponent)>
    </Deprecated>
  </Deprecated>
</div>
`;
